rancher/distros-test-framework
# feat: add test runner and makefile adjustments

#### Proposed Changes ####

-feat: add test runner and makefile adjustments

The proposal is to add a test-runner to run make file local and in docker

Bellow you find the related tasks





#### Types of Changes ####
- Yes README and examples updated


#### Testing ####


Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

- YES 


   UpgradeSUC - https://jenkins.int.rancher.io/job/rke2-tests/view/rke2-patch-validation/job/rke2_validate_cluster/383/console

   Windows -https://jenkins.int.rancher.io/job/rke2-tests/view/rke2-patch-validation/job/rke2_validate_cluster/385/console


2. Verify code lint; we should not have errors.
- ok 
![Screenshot 2023-09-28 at 18 20 51](https://github.com/rancher/distros-test-framework/assets/124721967/3604e8bc-442a-4cc8-b9dc-df30bb3150ba)


3. Update the documentation if needed.
- ok

4. Update makefile and docker run if adding new tests.
- ok 

5. Run your tests at least 4 times with all configurations needed and possible.
- ok 
6. If needed test with different os types.
- ok 

#### Linked Issues ####
- https://github.com/rancher/qa-tasks/issues/802
- https://github.com/rancher/qa-tasks/issues/854


#### Further Comments ####

- Variables precedence order 

 First: code
 Second: CLI
 Third: .env
